core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -283,17 +291,26 @@ public static String subtractPaths(String path, String prefix) throws InvalidPat
    * @throws InvalidPathException when the path or prefix is invalid
    */
   public static boolean hasPrefix(String path, String prefix) throws InvalidPathException {
-    String[] pathComponents = getPathComponents(path);
-    String[] prefixComponents = getPathComponents(prefix);
```

Comment:
Previously, `getPathComponents` calls `cleanPath`, which "resolves dot components," i.e. `/a/b/../c` becomes `/a/c`. Your reimplementation should preserve this behaviour, or make sure paths have been cleaned up before passed to `hasPrefix`.

@jiacheliu3 watch out for this behavioural change.

Reply:
Yes you are right. Already removed this condition.

Reply:
Got it. I called `cleanPath` to normalize the path and prefix.